database/step/list_image.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2023 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package step
+
+import (
+	"database/sql"
+
+	"github.com/go-vela/types/constants"
+)
+
+// ListStepImageCount gets a list of all step images and the count of their occurrence from the database.
+func (e *engine) ListStepImageCount() (map[string]float64, error) {
+	e.logger.Tracef("getting count of all images for steps from the database")
+
+	// variables to store query results and return value
+	s := []struct {
+		Image sql.NullString
+		Count sql.NullInt32
+	}{}
+	images := make(map[string]float64)
+
+	// send query to the database and store result in variable
+	err := e.client.
+		Table(constants.TableStep).
+		Select("image", " count(image) as count").
```

Comment:
Why the space?
```suggestion
		Select("image", "count(image) as count").
```

Reply:
The space is intentional and my OCD required I add it to make the query look prettier 😅 

Without the space, the query looks like:

```sql
SELECT "image",count(image) as count FROM "steps"
```

With the space, the query looks like:

```sql
SELECT "image", count(image) as count FROM "steps"
```

Reply:
LOL. It was on purpose, so it works for me.